### PR TITLE
vtls/rustls: adapt to the updated rustls_version proto

### PR DESCRIPTION
**Pull request by bagder**
Combine this with #7948

### CODE REVIEW

**Comment by kevinburke** [comment]
LGTM

File: lib/vtls/rustls.c [hunk 1]
```diff
@@ -543,14 +543,20 @@ cr_close(struct Curl_easy *data, struct connectdata *conn,
   }
 }
 
+static size_t cr_version(char *buffer, size_t size)
+{
+  struct rustls_str ver = rustls_version();
+  return msnprintf(buffer, size, "%s", ver.data);
```
Comment (jsha) [inline lib/vtls/rustls.c:549]: This should use ``%.*s`, ver.len, ver.data`, since ver's len will probably be shorter than the buffer's size.

Comment (bagder) [reply lib/vtls/rustls.c:549]: The header file says it is fine to treat it as a C string though, so shouldn't a plain %s write the same output?

Comment (jsha) [reply lib/vtls/rustls.c:549]: The header comment for rustls_str says:

```
/**
 * A read-only view on a Rust `&str`. The contents are guaranteed to be valid
 * UTF-8. As an additional guarantee on top of Rust's normal UTF-8 guarantee,
 * a `rustls_str` is guaranteed not to contain internal NUL bytes, so it is
 * safe to interpolate into a C string or compare using strncmp. Keep in mind
 * that it is not NUL-terminated.
 *
 * The memory exposed is available as specified by the function
 * using this in its signature. For instance, when this is a parameter to a
 * callback, the lifetime will usually be the duration of the callback.
 * Functions that receive one of these must not dereference the data pointer
 * beyond the allowed lifetime.
 */
typedef struct rustls_str {
  const char *data;
  size_t len;
} rustls_str;
```

So it's safe to _interpolate_ into a C string, but not safe to use directly as a C string (which assumes NUL termination). I'm happy to improve the docs if I can make them clearer!

Comment (bagder) [reply lib/vtls/rustls.c:549]: I think it was just a problem with this user not reading what it actually says... :open_mouth:

**Comment by bagder** [comment]
Merged into #7948 already

**Pull request closed without merging**